app/src/main/java/com/example/moviewithfragments/request/MovieApiClient.kt

```diff
@@ -0,0 +1,141 @@
+    private val viewModelMovies: MutableLiveData<List<MovieModel>?>
+    var retrieveMoviesRunnable: RetrieveMoviesRunnable? = null
+    val movies: MutableLiveData<List<MovieModel>?>
+        get() = viewModelMovies
+
+    companion object {
+        var instance: MovieApiClient? = null
+            get() {
+                if (field == null) {
+                    field = MovieApiClient()
+                }
+                return field
+            }
+            private set
+    }
+
+    init {
+        viewModelMovies = MutableLiveData()
+    }
+
+    fun searchMoviesApi(query: String, year: Int, pageNumber: Int) {
+
+        if (retrieveMoviesRunnable != null) {
+            retrieveMoviesRunnable = null
+        }
+        val retrieveMoviesRunnable = RetrieveMoviesRunnable(query, year, pageNumber)
+        val retroFitHandler: Future<*> =
+            appExecutors.getInstance()?.getNetworkIO()!!.submit(retrieveMoviesRunnable)
+        appExecutors.getInstance()!!.getNetworkIO()?.schedule(Runnable {
+            @Override
+            fun run() {
+                retroFitHandler.cancel(true)
+
+            }
+        }, 5000, TimeUnit.MILLISECONDS)
+    }
+
+    inner class RetrieveMoviesRunnable(
+        private val query: String,
+        private val year: Int,
+        private val pageNumber: Int
+    ) :
+        Runnable {
+
+        var cancelRequest = false
+        override fun run() {
+            try {
+                if (query.length != 0) {
+                    val response = getMovies(query, pageNumber)?.execute()
+                    responseData(response)
+                } else {
+                    if (year == 0) {
+                        val response = getPopularMovies()?.execute()
+                        responseData(response)
+                    } else {
+                        val response = getLatestMovies(year)?.execute()
+                        responseData(response)
+                    }
+                }
+            } catch (e: IOException) {
+                e.printStackTrace()
+            }
+        }
+
+        fun responseData(response: Response<MovieSearchResponse?>?) {
+            if (cancelRequest) {
+                return
+            }
+            if (response!!.code() == 200) {
+                val list = response.body()!!.movies as java.util.ArrayList<MovieModel>?
+                if (pageNumber == 1) {
+                    viewModelMovies?.postValue(list as List<MovieModel>?)
+                } else {
+
+                    val currentMovies =
+                        viewModelMovies.value as java.util.ArrayList<MovieModel>?
+                    currentMovies?.clear()
+                    currentMovies?.addAll(list!!)
+                    viewModelMovies.postValue(currentMovies!!)
+
+                }
+
+            } else {
+                val error = response.errorBody().toString()
+                Log.v("Tag", "Error" + error)
+                viewModelMovies.postValue(null)
```

Comment:
Do not post the values to LiveData like this
Instead use callbacks with sealed classes to easily model the response.